backend/module/eCampApi/test/CampTest.php

```diff
@@ -0,0 +1,158 @@
+        $this->campType->setIsCourse(false);
+        $this->campType->setOrganization($this->organization);
+
+        $this->camp = new Camp();
+        $this->camp->setName('CampName');
+        $this->camp->setTitle('CampTitle');
+        $this->camp->setMotto('CampMotto');
+        $this->camp->setCampType($this->campType);
+        $this->camp->setCreator($this->user);
+        $this->camp->setOwner($this->user);
+
+        $this->getEntityManager()->persist($this->camp);
+        $this->getEntityManager()->persist($this->campType);
+        $this->getEntityManager()->persist($this->organization);
+        $this->getEntityManager()->flush();
+    }
+
+    public function testCampFetch() {
+        $this->dispatch("/api/camps/{$this->camp->getId()}", 'GET');
+
+        $this->assertResponseStatusCode(200);
+
+        $host = '';
+        $expectedResponse = <<<JSON
+            {
+                "id": "{$this->camp->getId()}",
+                "name": "CampName",
+                "title": "CampTitle",
+                "motto": "CampMotto",
+                "role": "manager",
+                "_embedded": {
+                    "creator": {
+                        "_links": {
+                            "self": {
+                                "href": "http://{$host}/api/users/{$this->user->getId()}"
+                            }
+                        }
+                    },
+                    "campType": {
+                        "id": "{$this->campType->getId()}",
+                        "name": "CampType",
+                        "isJS": false,
+                        "isCourse": false,
+                        "_embedded": {
+                            "organization": {
+                                "_links": {
+                                    "self": {
+                                        "href": "http://{$host}/api/organizations/{$this->organization->getId()}"
+                                    }
+                                }
+                            },
+                            "activityTypes": []
+                        },
+                        "_links": {
+                            "self": {
+                                "href": "http://{$host}/api/camp-types/{$this->campType->getId()}"
+                            }
+                        }
+                    },
+                    "campCollaborations": [],
+                    "periods": [],
+                    "activityCategories": []
+                },
+                "_links": {
+                    "self": {
+                        "href": "http://{$host}/api/camps/{$this->camp->getId()}"
+                    },
+                    "activities": {
+                        "href": "http://{$host}/api/activities?campId={$this->camp->getId()}"
+                    }
+                }
+            }
+JSON;
+
+        $this->assertEquals(json_decode($expectedResponse), $this->getResponseContent());
```

Comment:
Hier würde ich eher wie folgt vorgehen:
- $this->getResponseContext()  ->  json_decode()
- prüfen einzelner Werte in Response

Ist stabiler wenn neue Felder dazu kommen.

Reply:
Das war hier eigentlich genau meine Idee, dass wir pro Endpoint 1 Test haben der das ganze Objekt vergleicht.

Idee dahinter ist, dass Hinzufügen/Löschen von weiteren Feldern oder auch switch zwischen Embedded/Non-Embedded nicht einfach unbemerkt duchflutscht, sondern bewusst der Test (und hoffentlich auch Docu und evt. Frontend) manuell angepasst werden muss.

Ich seh aber das Problem: Wenn CampType ein neues Property bekommt dann wird man einige Tests anpassen müssen. Vielleicht wäre ein Trade-Off die Embeddeds nicht im Detail zu prüfen sondern nur deren Existence?